### PR TITLE
fix: pr labeller patterns

### PR DESCRIPTION
**Pull request by d0choa**
## ✨ Context

PR labeler patterns don't start at the beginning of the paths. This is making a few incorrect tags to be added. For example:

- `Documentation` in `tests/docs/...`
- `Refactor` in PR title: `perf: refactor clump`
- ...

## 🛠 What does this PR implement

Updated patterns starting with "^"

## 🚦 Before submitting

- [x] Do these changes cover one single feature (one change at a time)?
- [x] Did you read the [contributor guideline](https://opentargets.github.io/gentropy/development/contributing/#contributing-checklist)?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you make sure there is no commented out code in this PR?
- [x] Did you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards in PR title and commit messages?
- [x] Did you make sure the branch is up-to-date with the `dev` branch?
- [x] Did you write any new necessary tests?
- [x] Did you make sure the changes pass local tests (`make test`)?
- [x] Did you make sure the changes pass pre-commit rules (e.g `poetry run pre-commit run --all-files`)?

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -25,39 +25,39 @@ labels:
...
-      - "src/gentropy/datasource/.*"
+      - "^src/gentropy/datasource/.*"
   - label: "Step"
     files:
-      - "src/gentropy/[a-zA-Z]\\w+\\.py"
+      - "^src/gentropy/[a-zA-Z]\\w+\\.py"
   - label: "Feature"
-    title: "feat.*"
+    title: "^feat.*"
   - label: "Bug"
```
Comment (ireneisdoomed) [inline .github/labeler.yml:46]: I don't love this label. I think it makes sense when the title of the PR starts with `fix`, meaning that the changes intend to fix an existing problem.
For commit messages, I personally use `fix` quite often to address small fixes in the code I am developing (a typo, or test fixes). Unless I am using `fix` unconventionally, I'd omit this label.

**Review by ireneisdoomed — APPROVED** [review]
All makes sense, ty! See my comment

File: .github/labeler.yml (hunk 1 above)
Comment (d0choa) [reply .github/labeler.yml:46]: https://www.conventionalcommits.org/en/v1.0.0/#specification

Comment (d0choa) [reply .github/labeler.yml:46]: Typos are also a type of bug.

Comment (d0choa) [reply .github/labeler.yml:46]: do you prefer fix?

Comment (ireneisdoomed) [reply .github/labeler.yml]: Forget what I said. I thought labels were assigned from the commit messages, not the PR title. It makes sense like this. Sorry!

**Pull request merged**